### PR TITLE
Module 5/logging and error handling

### PR DESCRIPTION
**Pull request by Mazorakopf**

### CODE REVIEW

File: src/middleware/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const otherQueryParam = ['limit', 'orderBy', 'sort'];
+
+export const buildQuery = (paramAttrMap, allowedParamValue) => {
+    return (req, res, next) => {
+        const query = { condition: {}, other: {} };
+        const failed = [];
+
+        for (const [param, value] of Object.entries(req.query)) {
+            if (allowedParamValue !== undefined
```
Comment (Autushonak) [inline src/middleware/index.js:12]: ! allowedParamValue

File: src/group/dao.js [hunk 2]
```diff
@@ -15,29 +16,36 @@ export const findById = (groupId) => {
     });
 };
 
-export const findAll = () => {
-    return Group.findAll({ include: Permission });
+export const findByQuery = (query) => {
+    if (query.condition.name !== undefined) {
```
Comment (Autushonak) [inline src/group/dao.js:20]: !query.condition.name

File: src/middleware/index.js (hunk 1 above)
Comment (Mazorakopf) [reply src/middleware/index.js]: fixed

File: src/group/dao.js (hunk 2 above)
Comment (Mazorakopf) [reply src/group/dao.js]: fixed

File: src/server.js [hunk 3]
```diff
@@ -13,10 +14,12 @@ app.use(express.json());
 app.use(`/api${UserController.path}`, UserController.router);
 app.use(`/api${PermissionController.path}`, PermissionController.router);
 app.use(`/api${GroupController.path}`, GroupController.router);
+app.use(expressLogger);
```
[the remark this answers is not in the document]
Comment (Mazorakopf) [reply src/server.js]: Yes, you are right. I fixed the logging

**Pull request merged**